src/Task/RealMeSetupTask.php

```diff
@@ -276,45 +279,33 @@ private function validateEntityID($forEnv)
             );
         }
 
-        // make sure the entityID is a valid URL
-        $entityId = filter_var($entityId, FILTER_VALIDATE_URL);
-        if ($entityId === false) {
-            $this->errors[] = _t(
-                RealMeSetupTask::class . '.ERR_CONFIG_ENTITYID',
-                'The Entity ID (\'{entityId}\') must be https, not be \'localhost\', and must contain a valid ' .
-                    'service name and privacy realm e.g. https://my-realme-integration.govt.nz/p-realm/s-name',
-                array(
-                    'entityId' => $entityId
-                )
-            );
-
+        $validationMessage = _t(
+            RealMeSetupTask::class . '.ERR_CONFIG_ENTITYID',
+            'The Entity ID (\'{entityId}\') must be https, not be \'localhost\', and must contain a valid ' .
+                'service name and privacy realm e.g. https://my-realme-integration.govt.nz/p-realm/s-name',
+            ['entityId' => $entityId]
+        );
+        // make sure the entityID is a valid URL with HTTPS protocol
+        if (!ConstraintValidator::validate($entityId, [new Url(protocols: ['https']), new NotBlank()])->isValid()) {
+            $this->errors[] = $validationMessage;
             // invalid entity id, no point continuing.
             return;
         }
 
-        // check it's not localhost and HTTPS. and make sure we have a host / scheme
+        // check it's not localhost
         $urlParts = parse_url($entityId ?? '');
-        if ($urlParts['host'] === 'localhost' || $urlParts['scheme'] === 'http') {
-            $this->errors[] = _t(
-                RealMeSetupTask::class . '.ERR_CONFIG_ENTITYID',
-                'The Entity ID (\'{entityId}\') must be https, not be \'localhost\', and must contain a valid ' .
-                    'service name and privacy realm e.g. https://my-realme-integration.govt.nz/p-realm/s-name',
-                array(
-                    'entityId' => $entityId
-                )
-            );
-
+        if ($urlParts['host'] === 'localhost') {
+            $this->errors[] = $validationMessage;
             // if there's this much wrong, we want them to fix it first.
             return;
         }
 
         $path = ltrim($urlParts['path'] ?? '');
-        $urlParts = preg_split("/\\//", $path ?? '');
-
+        $pathParts = preg_split("/\\//", $path ?? '');
```

Comment:
Unrelated but renamed this so it's not reusing the same variable name as above, which was for something different.